[RFC] vim-patch:7.4.2259,7.4.2268,7.4.2318,7.4.2320

jamessan opened the pull request:
#### vim-patch:7.4.2259

Problem:    With 'incsearch' can only see the next match.
Solution:   Make CTRL-N/CTRL-P move to the previous/next match. (Christian
            Brabandt)

https://github.com/vim/vim/commit/4d6f32cbfbaf324ac4a25c0206a5db0e9f7a48f7


#### vim-patch:7.4.2268

Problem:    Using CTRL-N and CTRL-P for incsearch shadows completion keys.
Solution:   Use CTRL-T and CTRL-G instead.

https://github.com/vim/vim/commit/1195669f9e434fa9ab8b57ee9470bf951e4990b8


#### vim-patch:7.4.2318

Problem:    When 'incsearch' is not set CTRL-T and CTRL-G are not inserted as
            before.
Solution:   Move vim/vim#ifdef and don't use goto.

https://github.com/vim/vim/commit/349e7d94e6bbb253bb87adad9039f095128ab543


#### vim-patch:7.4.2320

Problem:    Redraw problem when using 'incsearch'.
Solution:   Save the current view when deleting characters. (Christian
            Brabandt) Fix that the '" mark is set in the wrong position. Don't
            change the search start when using BS.

https://github.com/vim/vim/commit/dda933d06c06c2792bd686d059f6ad19191ad30b

raichoo committed:
vim-patch:7.4.2259

Problem:    With 'incsearch' can only see the next match.
Solution:   Make CTRL-N/CTRL-P move to the previous/next match. (Christian
            Brabandt)

https://github.com/vim/vim/commit/4d6f32cbfbaf324ac4a25c0206a5db0e9f7a48f7

jamessan committed:
functests/legacy: Add lua version of test_search.vim

jamessan committed:
vim-patch:7.4.2268

Problem:    Using CTRL-N and CTRL-P for incsearch shadows completion keys.
Solution:   Use CTRL-T and CTRL-G instead.

https://github.com/vim/vim/commit/1195669f9e434fa9ab8b57ee9470bf951e4990b8

jamessan committed:
vim-patch:7.4.2318

Problem:    When 'incsearch' is not set CTRL-T and CTRL-G are not inserted as
            before.
Solution:   Move vim/vim#ifdef and don't use goto.

https://github.com/vim/vim/commit/349e7d94e6bbb253bb87adad9039f095128ab543

jamessan committed:
vim-patch:7.4.2320

Problem:    Redraw problem when using 'incsearch'.
Solution:   Save the current view when deleting characters. (Christian
            Brabandt) Fix that the '" mark is set in the wrong position. Don't
            change the search start when using BS.

https://github.com/vim/vim/commit/dda933d06c06c2792bd686d059f6ad19191ad30b

bfredl commented on test/functional/legacy/search_spec.lua:
```diff
@@ -0,0 +1,470 @@
+local eq = helpers.eq
+local eval = helpers.eval
+local feed = helpers.feed
+local nvim = helpers.funcs
```
better use `funcs = helpers.funcs` as `helpers.nvim` does exists.

jamessan commented:
QB failures are known `get_keymap` timeout and the kbtree release build issues.

jamessan commented:
Appveyor is the known job_spec failure...

jamessan committed:
ex_getln: Lint command_line_handle_key readability/fn_size

Create new functions to handle moving to the next incsearch match or
matching history index.
